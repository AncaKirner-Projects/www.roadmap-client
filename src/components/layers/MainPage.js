import React from 'react';
import PropTypes from 'prop-types';
import { Route, Switch } from 'react-router-dom';
import CardList from '../main/CardList';
import Cart from '../main/Cart';
import AddProduct from '../main/AddProduct';
import ProductDescription from '../main/ProductDescription';

const MainPage = props => (
  <main className={props.classes.content}>
    <div className={props.classes.toolbar} />
    <Switch>
      <Route exact strict path="/" component={CardList} />
      <Route exact strict path="/products/add" component={AddProduct} />
      <Route exact strict path="/products/:id" component={ProductDescription} />
      <Route exact strict path="/cart" component={Cart} />
    </Switch>
    {/* <CardList /> */}
  </main>
);


MainPage.propTypes = {
  classes: PropTypes.object.isRequired
};

export default MainPage;
