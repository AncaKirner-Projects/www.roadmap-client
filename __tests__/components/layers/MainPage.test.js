import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MainPage from '../../../src/components/layers/MainPage';

describe('<NavBar />', () => {
  const props = {
    classes: {
      content: {},
      toolbar: {}
    }
  };
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<MainPage classes={props.classes} />);
  });

  it('renders 1 component', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('renders 1 <AppBar/> component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
