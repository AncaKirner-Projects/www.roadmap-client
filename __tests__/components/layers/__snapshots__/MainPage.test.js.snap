// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders 1 <AppBar/> component 1`] = `
<main
  className={Object {}}
>
  <div
    className={Object {}}
  />
  <Connect(CardList) />
</main>
`;
